# Домашнее задание по теме «Управляющие конструкции и коллекции» часть 1

## Задание 1

In [2]:
word = 'test'
word_len = len(word)

# ОПРЕДЕЛЯЕМ ПАРАМЕТРЫ ИНДЕКСАЦИИ
# для чётного числа
ind_start_even = int((word_len - 2)/2) # находим индекс начала
ind_end_even = ind_start_even+2        # находим индекс окончания

# для НЕчётного числа
ind_start_odd = int((word_len - 1)/2)  # находим индекс начала
ind_end_odd = ind_start_odd+1          # находим индекс окончания 

# УСЛОВНАЯ КОНСТРУКЦИЯ

if word_len % 2 == 0:
    print(word[ind_start_even:ind_end_even])
elif word_len % 2 != 0:
    print(word[ind_start_odd:ind_end_odd])
else: print('Хамяк его Знает')


es


## Задание № 2

In [56]:
number_from_user = input('Введите число: ')               # просим пользователя ввести значение
list_user_numbers = []                                  # список, в который будут записаны только числа

while number_from_user != 0:                            # начинаем цикл. условие такое, что цикл никогда не закончится
    try:
        number_from_user = int(number_from_user)        # пробуем перевести данные пользователя в тип даннзх "число"
        
    except Exception:
        print('Необходимо вводить только цифры')            # если это не получается, то выводим предупреждение и 
        number_from_user = input('Повторите: ')          # запускаем запрос данных заново

    else:                                               # если тип данных "число", то 
        list_user_numbers.append(number_from_user)      # заносим значение в список
        if number_from_user == 0:                       # если пользователь ввёл "0", то
            print(sum(list_user_numbers))               # выводим сумму всех значений
        else:
            number_from_user = input('Введите число: ')

Введите число: 1
Введите число: 2
Введите число: 3
Введите число: 4
Введите число: 5
Введите число: 0
15


## Задание № 3

### Сценарий № 1

In [20]:
# Входящие значения
boys = ['Peter', 'Alex', 'John', 'Arthur', 'Richard']
girls = ['Kate', 'Liza', 'Kira', 'Emma', 'Trisha']

# Сортировка списков
s_boys = sorted(boys)
s_girls = sorted(girls)

# Поиск минимальной длины списка для цикла
l_boys = len(boys)
l_girls = len(girls)
min_len = min(l_boys, l_girls)

# Проверка списков на равность
len_check = l_boys == l_girls

# Условие проверяет, равны ли списки
if len_check:
    # Если списки равны, запускается цикл
    for i in range(min_len):
        print(s_boys[i], ' и ', s_girls[i])
    # Если списки не равны, пользователь получает уведомление
else:
    print('Внимание, кто-то может остаться без пары!')


Alex  и  Emma
Arthur  и  Kate
John  и  Kira
Peter  и  Liza
Richard  и  Trisha


### Сценарий № 2

In [19]:
boys = ['Peter', 'Alex', 'John', 'Arthur', 'Richard', 'Michael']
girls = ['Kate', 'Liza', 'Kira', 'Emma', 'Trisha']

# Сортировка списков
s_boys = sorted(boys)
s_girls = sorted(girls)

# Поиск минимальной длины списка для цикла
l_boys = len(boys)
l_girls = len(girls)
min_len = min(l_boys, l_girls)

# Проверка списков на равность
len_check = l_boys == l_girls

# Условие проверяет, равны ли списки
if len_check:
    # Если списки равны, запускается цикл
    for i in range(min_len):
        print(s_boys[i], ' и ', s_girls[i])
    # Если списки не равны, пользователь получает уведомление
else:
    print('Внимание, кто-то может остаться без пары!')

Внимание, кто-то может остаться без пары!


## Задание № 4

In [51]:
# На входе получаем следующий список:
countries_temperature = [
    ['Thailand', [75.2, 77, 78.8, 73.4, 68, 75.2, 77]],
    ['Germany', [57.2, 55.4, 59, 59, 53.6]],
    ['Russia', [35.6, 37.4, 39.2, 41, 42.8, 39.2, 35.6]],
    ['Poland', [50, 50, 53.6, 57.2, 55.4, 55.4]]
]

# Подсчитываем кол-во стран. Полученное кол-во будем использовать в цикле
number_of_countries = len(countries_temperature)

for i in range(number_of_countries):
    # количество записей с температурами:
    kolvo_zap = len(countries_temperature[i][1])
    # сумма записей:
    summa_chisel = sum(countries_temperature[i][1])
    # находим среднее:
    srednee = summa_chisel / kolvo_zap
    # конвертируем град. Фаренгейта в град. Цельсия:
    perevod_c = (srednee - 32) / 1.8
    # округляем до одного символая после запятой:
    okruglenie = round(perevod_c, 1)
    
    print(countries_temperature[i][0], ' - ', okruglenie, ' C')


Thailand  -  23.9  C
Germany  -  13.8  C
Russia  -  3.7  C
Poland  -  12.0  C


#### Вопрос к заданию № 4

Прошу подсказать каким образом можно записать значения в список?
Меня интересуют 2 сценария:
    1.[[Thailand, Germany, Russia, Poland], [23.9, 13.8, 3.7, 12.0]]
    2.[23.9, 13.8, 3.7, 12.0]
    
Я попробовал применить методы:

a = [i for i in range(7)], но либо не работает, либо выдаёт не те результаты

In [ ]:
# пример
# если честно, то мне крайне сложно получать предсказуемый результат

for i in range(number_of_countries):
    primer = [sum(x) for x in countries_temperature[i][1]]
    print(primer)